### ISSUE TITLE
resolveCiFriendliesOnly parses the version of dependency in profile when it is managed by dependencyManagement

### ISSUE THREAD

**Issue by yangnuoyu** [mojohaus/flatten-maven-plugin#156]
I was testing the resolveCiFriendliesOnly mode when I had a pom with <dependency> in <profile> whose version was managed by <dependencyManagement> of its parent pom. The results showed that the <version> of <dependency> will be parse into it when the <profile> is active. Here is the example:
In the parent pom:
```  
    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>javax.annotation</groupId>
                <artifactId>javax.annotation-api</artifactId>
                <version>1.3.2</version>
            </dependency>
        </dependencies>
    </dependencyManagement>
```

In the child pom:
``` 
    <profiles>
        <profile>
            <id>test</id>
            <activation>
                <activeByDefault>true</activeByDefault>
            </activation>
            <dependencies>
                <dependency>
                    <groupId>javax.annotation</groupId>
                    <artifactId>javax.annotation-api</artifactId>
                </dependency>
            </dependencies>
        </profile>
    </profiles>
```

What I expected is that, the flattened pom will keep the <profiles> element unchanged but the result is as this:
```
  <profiles>
    <profile>
      <id>test</id>
      <activation>
        <activeByDefault>true</activeByDefault>
      </activation>
      <dependencies>
        <dependency>
          <groupId>javax.annotation</groupId>
          <artifactId>javax.annotation-api</artifactId>
          <version>1.3.2</version>
          <scope>compile</scope>
        </dependency>
      </dependencies>
    </profile>
  </profiles>
```
This may be a similar problem of #103 but I do not see a similar case like this. Since resolveCiFriendliesOnly mode only resolves variables revision, sha1 and changelist and keeps everything else, I think it should be a problem.

**Comment by salehsquared** [comment]
This issue appears to be fixed by PR #152.

**Comment by slawekjaranowski** [comment]
Mentioned issue and PR are closed / merged.
I assume that is fixed. 
If no - please reopen.
